prov/sm2/src/sm2_coordination.c

```diff
@@ -0,0 +1,675 @@
+/*
+ * Copyright (c) 2023 Amazon.com, Inc. or its affiliates.
+ * All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "sm2_coordination.h"
+#include "sm2.h"
+#include "sm2_atom.h"
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#define NEXT_MULTIPLE_OF(x, mod) x % mod ? ((x / mod) + 1) * mod : x
+#define ZOMBIE_ALLOCATION_NAME "ZOMBIE"
+
+static void sm2_coordinator_attempt_shm_file_shrink(struct sm2_mmap *map);
+
+/**
+ * @brief take an open file, and mmap its contents
+ *
+ * @param[in] fd
+ * @param[out] map
+ */
```

Comment:
I really do not want to do this.  The function description you pointed out is not very useful, but I think the majority of comments (both function description and in-code comments) are incredibly useful for understanding.  I understand that there are different perspectives on the value of comments.  Comments are often the first part of code to rot.  But I think these are super useful for what is going on (not all, but most).  I would like to be able to keep the useful comments in the code, and I will happily remove the ones that aren't useful.

Reply:
Comments are detrimental to maintenance, outside of published APIs.  It's another area of the code that MUST be changed when the source is changed -- a change that is frequently skipped -- resulting in comments that are not aligned with the code, and is most often used to avoid designing a function call that is obvious in its use, or which tries to do way more than it should.  Make the functions names clear as to what they do, with arguments that are obvious.  Use const to distinguish between input and output pointers.

Comments should be for things that aren't obvious by reading the code, like lock ordering.